assets/js/theme/common/carousel/utils/handleImageLoad.js

```diff
@@ -3,18 +3,43 @@ import getActiveSlideInfo from './getActiveSlideInfo';
 const IMAGE_ERROR_CLASS = 'is-image-error';
 const IS_ANALYZED_DATA_ATTR = 'image-load-analyzed';
 
+const generateImage = ($slide, $image) => {
+    $('<img />')
+        .on('error', () => $slide.addClass(IMAGE_ERROR_CLASS))
+        .attr({
+            src: $image.attr('src'),
+            srcset: $image.attr('srcset'),
+            sizes: $image.attr('sizes'),
+        });
+};
+
 export default (e, carousel) => {
     const {
         isAnalyzedSlide,
-        attrsObj,
         $activeSlide,
+        $activeSlideImg,
+        activeSlideImgNode,
     } = getActiveSlideInfo(carousel, IS_ANALYZED_DATA_ATTR);
 
     if (isAnalyzedSlide) return;
 
     $activeSlide.data(IS_ANALYZED_DATA_ATTR, true);
 
-    $('<img />')
-        .on('error', () => $activeSlide.addClass(IMAGE_ERROR_CLASS))
-        .attr(attrsObj);
+    if (activeSlideImgNode.complete) {
+        if (activeSlideImgNode.naturalHeight === 0) {
+            $activeSlide.addClass(IMAGE_ERROR_CLASS);
+        } else if (activeSlideImgNode.naturalHeight === 1) {
+            generateImage($activeSlide, $activeSlideImg);
+        }
+
+        return;
+    }
+
+    // if IE
+    if (document.documentMode) {
```

Comment:
as a suggestion: we have IE helper -  isBrowserIE (path:assets/js/theme/common/utils/ie-helpers.js). So if `document.documenMode` property is more universal in terms of use, just update this helper and you can use it here without a comment.

Reply:
Updated helper with my implementation